ffi-convert-derive/src/utils.rs

```diff
@@ -128,31 +121,31 @@ pub fn parse_field(field: &syn::Field) -> Field {
 /// `(std::module1::module2::Vec`, `Hello`)`
 ///
 pub fn generic_path_to_concrete_type_path(
-    path: syn::TypePath,
+    mut path: syn::TypePath,
 ) -> (syn::TypePath, Option<syn::AngleBracketedGenericArguments>) {
-    let mut path_mut = path.clone();
-    let last_seg: Option<&mut syn::PathSegment> = path_mut.path.segments.last_mut();
+    let last_seg: Option<&mut syn::PathSegment> = path.path.segments.last_mut();
 
     if let Some(last_segment) = last_seg {
         if let syn::PathArguments::AngleBracketed(ref bracketed_type_params) =
             last_segment.arguments
         {
             let extracted_type_params = (*bracketed_type_params).clone();
             last_segment.arguments = syn::PathArguments::None;
-            (path_mut, Some(extracted_type_params))
+            (path, Some(extracted_type_params))
         } else {
-            (path_mut, None)
+            (path, None)
         }
     } else {
-        panic!("The field with type : {:?} is invalid. No segments on the TypePath")
+        panic!("Invalid type path: no segments on the TypePath")
```

Comment:
Too bad we're losing field type info here ! Proc-macros are very hard to debug. Any bit of information is good to take :) !

Reply:
Yup this should definitely be added sometime (I didn't change it there as we already don't have the info)